Jenkinsfile.trigger

```diff
@@ -161,12 +181,47 @@ for (buildObj in queue) {
 				quietPeriod: 5, // seconds
 			)
 			// TODO do something useful with "res.result" (especially "res.result != 'SUCCESS'")
-			// (maybe store "res.startTimeInMillis + res.duration" as endTime so we can implement some amount of backoff somehow?)
 			echo(res.result)
 			if (res.result != 'SUCCESS') {
+				def c = 1
+				if (pastFailedJobs[buildObj.buildId]) {
+					c += pastFailedJobs[buildObj.buildId].count
+				}
+				// TODO maybe implement some amount of backoff? keep first url/endtime?
+				newFailedJobs[buildObj.buildId] = [
+					count: c,
+					identifier: identifier,
+					url: res.absoluteUrl,
+					endtime: (res.startTimeInMillis + res.duration) / 1000.0, // convert to seconds
+				]
+
 				// "catchError" is the only way to set "stageResult" :(
 				catchError(message: 'Build of "' + identifier + '" failed', buildResult: 'UNSTABLE', stageResult: 'FAILURE') { error() }
 			}
 		}
 	}
 }
+
+// save newFailedJobs so we can use it next run as pastFailedJobs
+node {
+	def newFailedJobsJson = writeJSON(json: newFailedJobs, returnText: true)
+	withEnv([
+		'newFailedJobsJson=' + newFailedJobsJson,
+	]) {
+		stage('Archive') {
+			dir('builds') {
+				deleteDir()
+				sh '''#!/usr/bin/env bash
+					set -Eeuo pipefail -x
+
+					jq '.' <<<"$newFailedJobsJson" | tee pastFailedJobs.json
+				'''
+				archiveArtifacts(
+					artifacts: '*.json',
+					fingerprint: true,
+					onlyIfSuccessful: true,
```

Comment:
This confusingly refers to the success of _this_ job, not the success of all the jobs this job is triggering (nor all the stages of this job, which each have their own status).  This is a nice way to give us a safety against losing our data since we're doing a very poor-man's linked list here and one hiccup means we lose our intermediate data. :smile:

(Hence pulling the previous version of this file from the `lastSuccessful` instead of just the most recent job.)